lib/ddtrace/contrib/http/instrumentation.rb

```diff
@@ -67,14 +67,7 @@ def request(req, body = nil, &block) # :yield: +response+
                 span.resource = req.method
 
                 if pin.tracer.enabled && !Datadog::Contrib::HTTP.should_skip_distributed_tracing?(pin)
-                  req.add_field(Datadog::Ext::DistributedTracing::HTTP_HEADER_TRACE_ID, span.trace_id)
-                  req.add_field(Datadog::Ext::DistributedTracing::HTTP_HEADER_PARENT_ID, span.span_id)
-                  if span.context.sampling_priority
-                    req.add_field(
-                      Datadog::Ext::DistributedTracing::HTTP_HEADER_SAMPLING_PRIORITY,
-                      span.context.sampling_priority
-                    )
-                  end
+                  Datadog::HTTPPropagator.inject!(span.context, req)
```

Comment:
If this works, this seems good to me. Also this seems to fix B3 for Net::HTTP?

An important caveat though: `inject!` expects an `Env` and not a request object. I think we get away with this because of duck typing and the use of `[]=` between inputs seems to be consistent thus far.

However, we should be careful because it means our `HTTPPropagator` must now always use `[]=` lest it break the `Net::HTTP` integration in an obscure way. The basic flaw here is that `HTTPPropagator` is making strong assumptions about the behavior of its input without constraining or communicating that assumption. This is not a flaw in your changes, but your changes do reveal this latent flaw.

Non-action item, but just food for thought: we could modify the `HTTPPropagator` to be more flexible/agnostic to the payloads its injecting into. e.g. something like:

```
# Pass a block that takes a header and value.
# Block should add the name/value to add the header in
# a manner appropriate to the transport library being used.
Datadog::HTTPPropagator.inject!(span.context) do |name, value|
  # e.g. Net::HTTP compatible
  req[name] = value
  # OR Interface for a different transport client?
  req.add_field(name, value)
end
```